app/lib/actions/katello/alternate_content_source/update.rb

```diff
@@ -19,14 +19,14 @@ def plan(acs, smart_proxies, products, acs_params)
           products_to_associate = []
           products_to_disassociate = []
 
-          if acs.simplified?
-            products = products.uniq
-            products_to_associate = products - acs.products
-            products_to_disassociate = acs.products - products
-            old_product_ids = acs.products.pluck(:id)
-            acs.products = products
-            acs.audit_updated_products(old_product_ids) unless products_to_associate.empty? && products_to_disassociate.empty?
-          end
+          products = products.uniq
```

Comment:
We should still have an `if` statement, how about a check first if products were given at all? 

```ruby
if products.present?
...
```

Reply:
Would you mind explaining a little more about what this is accomplishing?

Reply:
Yeah, so as you have it, the code to calculate the products to add and remove will be run regardless of  whether or not the product list is empty.  That means those CPU cycles are being wasted.

The code leading up to `acs.save!` could be put in an `if` statement that checks if there are any products at all.

Reply:
The specific code block:

```ruby
          products = products.uniq
          products_to_associate = products - acs.products
          products_to_disassociate = acs.products - products
          old_product_ids = acs.products.pluck(:id)
          acs.products = products
          acs.audit_updated_products(old_product_ids) unless products_to_associate.empty? && products_to_disassociate.empty?
```